dist/spec/verifiable-payid-protocol-NC.txt

```diff
@@ -0,0 +1,896 @@
+   server has full control over the contents of the response message,
+   and may go rogue or be compromised.  The PayID owner has no way of
+   knowing if the PayID server behaves maliciously.  This implicit trust
+   assumption between the PayID owner and PayID server is not ideal in
+   the non-custodial settings.
+
+   The motivation for this extension of verifiable PayID protocol is to
+   enable non-custodial service providers to provide non-custodial PayID
+   service while preserving their existing security model.  More
+   specifically, this extension allows PayID owners to digitally sign
+   the payment account(s) information resource locally on their
+   application/device with their PayID private key (which never leaves
+   their device).  This signed payment account(s) information resource
+   can then be securely transferred to the non-custodial PayID service
+   providers' servers who can then send this to the PayID client in
+   response to a PayID query along with PayID owner's "identity"
+   information.  The PayID client and/or the sender of the transaction
+   can use this information to verify if the payment account(s) address
+   information is signed by the PayID owner and decide if they want to
+   proceed with the transaction.
+
+3.  Verifiable PayID Protocol Extension Specification
+
+   The Verifiable PayID protocol is designed along the same design
+   principles as [PAYID-PROTOCOL].
+
+3.1.  PaymentInformation Resource as JSON Web Signatures
+
+   We describe the payment accounts(s) information resource per
+   [PAYID-PROTOCOL] that is digitally signed as "PaymentInformation"
+   resource represented as a signed JSON Web Tokens (JWT) [RFC7519]
+   known as JSON Web Signature (JWS) [RFC7515].  JWT is a compact, URL-
+   safe means of representing claims to be transferred between two
+   parties.  JWTs encode claims as a JSON object that is used as the
+   payload of a JWS structure enabling the claims to be digitally
+   signed.  Below we define the three components of JWS.
+
+3.1.1.  JOSE Protected Header
+
+   For JWS, the members of the JSON object represented by the JOSE
+   Header describe the cryptographic operations applied to the JWS
+   Protected header and the JWS payload and optionally additional
+   properties of the JWS.
+
+
+
+Malhotra & Schwartz     Expires January 31, 2021                [Page 4]
+
+Internet-Draft     Verifiable PayID Protocol Extension         July 2020
+
+
+   For a complete list of members of this object, refer to [RFC7519] and
+   [RFC7515].  Following is an example of a JWT object representing JOSE
+   header parameters for JWS JSON Serialization syntax.
+
+   {
+          "name": "identityKey",
+          "alg" : "ES256K",
+          "typ" : "JOSE+JSON",
+          "b64" : "false",
+          "crit": ["b64"],
+          "jwk" :  {
+               "kty": "EC",
+               "use": "sig",
+               "crv": "secp256k1",
+               "x"  : "0",
+               "y". : "0",
```

Comment:
```suggestion
               "y". : "0",
               "kid": "9eb5ce46-4240-4ba7-b39a-317534ad3f06",
               "alg": "ES256K"
```

Not sure if we _also_ want to limit the key to signing operations, but that is possible by adding the alg to the `jwk`. Not sure if this is advisable or not?